packages/grid/x-data-grid-pro/src/tests/rows.DataGridPro.test.tsx

```diff
@@ -442,23 +444,23 @@ describe('<DataGridPro /> - Rows', () => {
         />,
       );
 
-      const virtualScroller = document.querySelector('.MuiDataGrid-virtualScroller')!;
-      const renderingZone = document.querySelector<HTMLElement>(
-        '.MuiDataGrid-virtualScrollerRenderZone',
-      )!;
+      const root = grid('root')!;
+      const virtualScroller = grid('virtualScroller')!;
+      const renderingZone = grid('virtualScrollerRenderZone')!;
       virtualScroller.scrollTop = 10e6; // scroll to the bottom
       act(() => virtualScroller.dispatchEvent(new Event('scroll')));
 
-      const lastCell = document.querySelector('[role="row"]:last-child [role="cell"]:first-child')!;
+      const lastCell = $('[role="row"]:last-child [role="cell"]:first-child')!;
       expect(lastCell).to.have.text('995');
       expect(renderingZone.children.length).to.equal(
         Math.floor((height - 1) / rowHeight) + rowBuffer,
       ); // Subtracting 1 is needed because of the column header borders
+      const scrollbarSize = apiRef.current.state.dimensions.scrollbarSize;
       const distanceToFirstRow = (nbRows - renderingZone.children.length) * rowHeight;
-      expect(renderingZone.style.transform).to.equal(
-        `translate3d(0px, ${distanceToFirstRow}px, 0px)`,
-      );
-      expect(virtualScroller.scrollHeight).to.equal(nbRows * rowHeight);
+      const styles = getComputedStyle(root);
+      const offsetTop = parseInt(styles.getPropertyValue('--DataGrid-offsetTop'), 10);
```

Comment:
Tbh a lot of these tests were too implementation-specific and they still are, it's just that finding a way to test what we really want to test (that the content is rendered where it needs to be) is kinda hard without falling into this trap. Something like argos tests but that can take snapshots at different moments of an action/test would be more reliable. This would be the best option to test resizing also.

Reply:
Or a video-based argos instead of image-based.

Reply:
Fair enough